ibis/backends/base/sql/alchemy/__init__.py

```diff
@@ -125,6 +126,14 @@ class BaseAlchemyBackend(BaseSQLBackend):
     supports_temporary_tables = True
     _temporary_prefix = "TEMPORARY"
 
+    def _from_url(self, url: str, **kwargs) -> BaseBackend:
+        """Construct an ibis backend from a SQLAlchemy-conforming URL."""
+        new_kwargs = kwargs.copy()
+        kwargs = {}
+        _, new_kwargs = self.inspector.dialect.create_connect_args(url)
+        new_kwargs = toolz.merge(kwargs, new_kwargs)
+        return self.connect(**new_kwargs)
+
```

Comment:
Hi @cpcloud and @gforsyth,

I stumbled upon this, while fixing issues with the `Exasol` backend.
I think this implementation should be more robust for sqla based backends, because SQLA dialects may adapt `url` specifics ([see here](https://docs.sqlalchemy.org/en/20/core/internals.html#sqlalchemy.engine.default.DefaultDialect.create_connect_args)).

For example, see [asyncpg](https://github.com/sqlalchemy/sqlalchemy/blob/4da59c6f534371d8c76cd728e8a459018e9112b6/lib/sqlalchemy/dialects/postgresql/asyncpg.py#L1106) and other dialects.

If you wan't, I can create an extra PR for just this change. If you think the change is not worth it, I'll drop the change in this PR too.

Reply:
HI @Nicoretti !  We certainly welcome improvements to make implementations more robust, and calling out to `create_connect_args` seems like a good idea.
We also have some extra handling for reading from environment variables in https://github.com/ibis-project/ibis/blob/master/ibis/backends/base/sql/__init__.py#L37-L70 that we don't want to override completely.

(the fact that sqlalchemy-specific code is included in `base/sql` is a separate problem that we need to fix)

Reply:
Hi @gforsyth, 

Thanks for the feedback. I'll omit it from this pull request and will  propose a more comprehensive change in a separate pull request in the next few days.